src/inmanta/deploy/state.py

```diff
@@ -472,13 +492,10 @@ def should_skip_for_dependencies(self, resource: "ResourceIdStr") -> bool:
         :param resource: The id of the resource to find the dependencies for
         """
         dependencies: Set[ResourceIdStr] = self.requires.get(resource, set())
-        return any(
-            # Determine based on latest deploy result rather than compliance status, because compliance status is unstable
-            # (e.g. HAS_UPDATE).
-            # Make sure to not skip on NEW (never deployed) resources, because they will not generate a discovery event.
-            self.resource_state[dep_id].last_deploy_result not in (DeployResult.DEPLOYED, DeployResult.NEW)
-            for dep_id in dependencies
-        )
+        for dep_id in dependencies:
```

Comment:
Slightly preferred the `any` approach for readability, but not really a change request.